### ISSUE TITLE
Adding 'print' javascript to an existing PDF file

### ISSUE THREAD

**Issue by amitkanfer** [galkahana/PDF-Writer#55]
Hi Gal,
Very nice project, thanks!

Can you please guide me what's the best and easiest way to make an existing PDF file to be 'auto-printed' when it's downloaded to a browser?

Thanks a lot!

Amit

**Comment by KarlHegbloom** [comment]
You should not auto print onto paper. It's their paper,  not yours. Let the
person downloading decide whether to print or to read it in a pdf reader.

**Comment by amitkanfer** [comment]
:)
Seriously,
Let me rephrase - automatically open the print dialog.

**Comment by galkahana** [comment]
Nice one.   
Did some looking around and it turns out that you can actually run javascript code as the document opens. and running something like this.print loads the dialog. at least when opened in acrobat...you can try other envs.

[package.zip](https://github.com/galkahana/PDF-Writer/files/384816/package.zip)

the key things to do are:

1. Create Javascript Action object with "this.print()" as its command
2. Attach this action object to the document root (a.k.a catalog) as its "OpenAction". This is done by creating modified version of the document root with this "OpenAction" key added.

It's just as possible to do so also with new documents, and not modified ones. In that case step 1 remains the same. Step 2 is changed to simply adding the key while hummus writes the document root by listeting to the `OnCatalogWrite` event like this:

```` 
    pdfWriter.getEvents().on('OnCatalogWrite',function(params) {
		params.catalogDictionaryContext
		    .writeKey('OpenAction')
		    .writeObjectReferenceValue(actionObjectID);
    });
````
